packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+  KomenciSessionInvalidError,
+  PepperNotCachedError,
+  storeTimestampIfKomenciError,
+} from 'src/identity/feelessVerificationErrors'
+import {
+  e164NumberToSaltSelector,
+  E164NumberToSaltType,
+  feelessAcceptedAttestationCodesSelector,
+  feelessAttestationCodesSelector,
+  FeelessVerificationState,
+  feelessVerificationStateSelector,
+  isFeelessVerificationStateExpiredSelector,
+} from 'src/identity/reducer'
+import { startAutoSmsRetrieval } from 'src/identity/smsRetrieval'
+import { VerificationStatus } from 'src/identity/types'
+import {
+  AttestationCode,
+  attestationCodeReceiver,
+  completeAttestations,
+  getActionableAttestations,
+  getAttestationsStatus,
+  NUM_ATTESTATIONS_REQUIRED,
+  requestAndRetrieveAttestations,
+  revealAttestations,
+  VERIFICATION_TIMEOUT,
+} from 'src/identity/verification'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import Logger from 'src/utils/Logger'
+import { setMtwAddress } from 'src/web3/actions'
+import { getContractKit } from 'src/web3/contracts'
+import { registerWalletAndDekViaKomenci } from 'src/web3/dataEncryptionKey'
+import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
+import { TransactionReceipt } from 'web3-eth'
+
+const TAG = 'identity/feelessVerification'
+
+const KOMENCI_URL = 'https://komenci.celo-networks-dev.org'
+// const KOMENCI_URL = 'http://192.168.86.33:3000'
+// TODO: Populate this with expected implementation address
+const ALLOWED_MTW_IMPLEMENTATIONS: Address[] = ['0x88a2b9B8387A1823D821E406b4e951337fa1D46D']
+const CURRENT_MTW_IMPLEMENTATION_ADDRESS: Address = '0x88a2b9B8387A1823D821E406b4e951337fa1D46D'
+
+export function* feelessFetchVerificationState() {
+  Logger.debug(TAG, '@feelessFetchVerificationState', 'Starting fetch')
+  try {
+    ValoraAnalytics.track(VerificationEvents.verification_fetch_status_start)
+    const [contractKit, walletAddress, feelessVerificationState, e164Number]: [
+      ContractKit,
+      string,
+      FeelessVerificationState,
+      string
+    ] = yield all([
+      call(getContractKit),
+      call(getConnectedUnlockedAccount),
+      select(feelessVerificationStateSelector),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.GettingStatus)),
+    ])
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    try {
+      // Checks if a MTW was verified in a previous attempt and updates state.
+      // This check will likely fail because the pepper hasn't been cached yet
+      // but needs to happen for the edge case that a user has a cached pepper
+      // and Komenci is down
+      yield call(fetchVerifiedMtw, contractKit, walletAddress, e164Number)
+    } catch (error) {
+      Logger.debug(TAG, 'Unable to check it MTW is verified on first attempt')
```

Comment:
Typo "it MTW is" -> "if MTW is"
```suggestion
      Logger.debug(TAG, 'Unable to check if MTW is verified on first attempt')
```